content/en/blog/_posts/2024-12-11-volume-group-snapshot-beta.md

```diff
@@ -0,0 +1,373 @@
+quiesce can be too expensive so you want to do it less frequently.
+For example, a user may want to run weekly backups with application quiesce
+and nightly backups without application quiesce but with consistent group support which
+provides crash consistency across all volumes in the group.
+
+## Kubernetes Volume Group Snapshots API
+
+Kubernetes Volume Group Snapshots introduce [three new API
+objects](https://github.com/kubernetes-csi/external-snapshotter/tree/master/client/apis/volumegroupsnapshot)
+for managing snapshots:
+
+`VolumeGroupSnapshot`
+: Created by a Kubernetes user (or perhaps by your own automation) to request
+creation of a volume group snapshot for multiple persistent volume claims.
+It contains information about the volume group snapshot operation such as the
+timestamp when the volume group snapshot was taken and whether it is ready to use.
+The creation and deletion of this object represents a desire to create or delete a
+cluster resource (a group snapshot).
+
+`VolumeGroupSnapshotContent`
+: Created by the snapshot controller for a dynamically created VolumeGroupSnapshot.
+It contains information about the volume group snapshot including the volume group
+snapshot ID.
+This object represents a provisioned resource on the cluster (a group snapshot).
+The VolumeGroupSnapshotContent object binds to the VolumeGroupSnapshot for which it
+was created with a one-to-one mapping.
+
+`VolumeGroupSnapshotClass`
+: Created by cluster administrators to describe how volume group snapshots should be
+created, including the driver information, the deletion policy, etc.
+
+These three API kinds are defined as CustomResourceDefinitions (CRDs).
+These CRDs must be installed in a Kubernetes cluster for a CSI Driver to support
+volume group snapshots.
+
+## What components are needed to support Volume Group Snapshots
+
+Volume group snapshots are implemented in the
+[external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter) repository.
+Implementing volume group snapshots meant adding or changing several components:
+
+* Added new CustomResourceDefinitions for VolumeGroupSnapshot and two supporting APIs.
+* Volume group snapshot controller logic is added to the common snapshot controller.
+* Adding logic to make CSI calls into the snapshotter sidecar controller.
+
+The volume snapshot controller and CRDs are deployed once per
+cluster, while the sidecar is bundled with each CSI driver.
+
+Therefore, it makes sense to deploy the volume snapshot controller and CRDs as a cluster addon.
+
+It is strongly recommended that Kubernetes distributors
+bundle and deploy the volume snapshot controller and CRDs as part
+of their Kubernetes cluster management process (independent of any CSI Driver).
+
+## What's new in Beta?
+
+* The VolumeGroupSnapshot feature in [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) moved to GA in the v1.11.0 release.
```

Comment:
Avoid hyperlinking to _master_ branches; these aren't stable in the way that permalinks are.

Is there a tag we can link to?

Reply:
Replaced with https://github.com/container-storage-interface/spec/releases/tag/v1.11.0.